integration/query_projection_compat_test.go

```diff
@@ -100,6 +102,28 @@ func TestQueryProjectionCompat(t *testing.T) {
 			filter:     bson.D{},
 			projection: bson.D{{"_id", true}, {"v", true}},
 		},
+		"Assign1Field": {
+			filter:     bson.D{},
+			projection: bson.D{{"foo", primitive.NewObjectID()}},
+		},
+		"AssignID": {
+			filter:      bson.D{},
+			projection:  bson.D{{"_id", primitive.Binary{Subtype: 0x80, Data: []byte{42, 0, 13}}}},
+			skipIDCheck: true,
+		},
+		"Assign1FieldIncludeID": {
+			filter:     bson.D{},
+			projection: bson.D{{"_id", true}, {"foo", primitive.NewDateTimeFromTime(time.Unix(0, 0))}},
+		},
+		"Assign2FieldsIncludeID": {
+			filter:     bson.D{},
+			projection: bson.D{{"_id", true}, {"foo", nil}, {"bar", "qux"}},
+		},
+		"Assign1FieldExcludeID": {
+			filter:      bson.D{},
+			projection:  bson.D{{"_id", false}, {"foo", primitive.Regex{Pattern: "^fo"}}},
+			skipIDCheck: true,
+		},
```

Comment:
Just noticed we don't have test case for testing exclusion and inclusion both. I think below two should error but give different error message. Could you add them please?
```
"Exclusion": {
			filter:     bson.D{},
			projection: bson.D{{"foo", false}, {"bar", true}},
		},
```

```
"Inclusion": {
			filter:     bson.D{},
			projection: bson.D{{"foo", true}, {"bar", false}},
		},
```

Reply:
Perfect thanks!